src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2646,10 +2633,21 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
                 BoundExpression operand,
                 Conversion conversion,
                 TypeWithAnnotations targetType,
-                TypeWithState operandType)
+                TypeWithState operandType,
+                bool isReachable)
             {
-                return ApplyConversion(
-                    node,
+                bool previousDisabledDiagnostics = _disableDiagnostics;
+                // If the node is not reachable, then we're only visiting to get
+                // nullability information for the public API, and not to produce diagnostics.
+                // Disable diagnostics, and return default for the resulting state
```

Comment:
>return default for the resulting state [](start = 44, length = 38)

What's the benefit of returning `default` rather than the actual value from visiting? Does this choice (default vs. actual state) have observable effects?

Reply:
Yes. It's used in the `Join` after the call to `convertResult`. If this isn't reachable, it shouldn't have an impact on the final state of the conditional.

---
In reply to: [282224091](https://github.com/dotnet/roslyn/pull/35544#discussion_r282224091) [](ancestors = 282224091)